ext/standard/tests/math/round_gh12143_adjustment_result_digit.phpt

```diff
@@ -0,0 +1,10 @@
+--TEST--
+Fix GH-12143: Improved handling of adjusting result digits
+--FILE--
+<?php
+var_dump(round(1.700000000000145, 13, PHP_ROUND_HALF_UP));
+var_dump(round(-1.700000000000145, 13, PHP_ROUND_HALF_UP));
```

Comment:
could you maybe add more tests to cover more large number test cases and different rounding modes.